### DISCUSSION TITLE
defining electrode thickness as a function of temperature

### DISCUSSION THREAD

**Discussion by ahsadeghi** [pybamm-team/PyBaMM#4169] [Q&A]
Hello,
I am trying to define `Negative electrode thickness [m]` in the `get_parameter_values()` as  a function of temperature but I get the following error:

`TypeError: Cannot process parameter '<function anode_NMC_decom at 0x000001EEFE8ABBA0>'`

I wonder if you would please help me to solve the problem.

Regards,
Hosein

**Comment by TomTranter** [comment]
I don't think it will work as it changes the mesh. No quick and easy way round it to the best of my knowledge. @valentinsulzer would non dimensionalising the thickness be straightforward?

**Reply by valentinsulzer** [reply]
No, this would be very challenging, especially allowing it to change during the simulation as a function of temperature

**Reply by ahsadeghi** [reply]
Thank you for your reply.
